src/main/java/org/dependencytrack/tasks/RepositoryMetaAnalysisTask.java

```diff
@@ -144,9 +144,9 @@ private void dispatchComponents(final List<ComponentProjection> components) {
 
     private List<ComponentProjection> fetchNextComponentsPage(final PersistenceManager pm, final Project project, final long offset) throws Exception {
         try (final Query<Component> query = pm.newQuery(Component.class)) {
-            var filter = "project.active == :projectActive && purlCoordinates != null";
+            var filter = "project.inactiveSince == :projectInactiveSince && purlCoordinates != null";
             var params = new HashMap<String, Object>();
-            params.put("projectActive", true);
+            params.put("projectInactiveSince", null);
```

Comment:
`null` should be passed as a literal, see https://github.com/DependencyTrack/dependency-track/issues/2540

```suggestion
            var filter = "project.inactiveSince == null && purlCoordinates != null";
            var params = new HashMap<String, Object>();
```